include/wil/com_apartment_variable.h

```diff
@@ -0,0 +1,230 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_COM_APARTMENT_VARIABLE_INCLUDED
+#define __WIL_COM_APARTMENT_VARIABLE_INCLUDED
+
+#include <unordered_map>
+#include <any>
+#include <type_traits>
+#include <Unknwn.h>
+#include <winrt/base.h>
+#include <roapi.h>
+#include <objidl.h>
+#include <wil/result_macros.h>
+
+#ifndef WIL_ENABLE_EXCEPTIONS
+#error This header requires exceptions
+#endif
+
+namespace wil
+{
+    // Determine if apartment variables are supported in the current process context.
+    // Prior to https://microsoft.visualstudio.com/OS/_git/os.2020/pullrequest/5750493 the

```

Comment:
Maybe add a description for non-Microsoft readers?

Reply:
RoRegisterForApartmentShutdown and the related APIs is what the comment is referring to.